service/redis/client.go

```diff
@@ -301,3 +303,18 @@ func (c *client) getConfigParameters(config string) (parameter string, value str
 	}
 	return s[0], strings.Join(s[1:], " "), nil
 }
+
+func (c *client) IsSyncing(ip, password string) (bool, error) {
```

Comment:
I would use a function `slaveIsReady` for the safety of update process. This can provide a wide overview of the cluster health checking the slaves were are more conditions than syncing state.

  1. master_host != 127.0.0.1  (pod bootstrap state, needs to be added to the cluster by operator)
  2. master_link_status:up
  3. master_sync_in_progress:0

Reply:
done :)